### PR TITLE
🔔 Create an event system

### LINKED ISSUE #57: 🔔 Create event system
# 👮‍♂️ What's needed
We shall create a route and a whole system to create events and enable users to fetch new events.

# 📜 Sources
```C
enum client_event {
    LOGED_IN,
    LOGED_OUT,
    DM_RECEIVED,
    TEAM_CREATED,
    CHANNEL_CREATED,
    THREAD_CREATED,
    THREAD_REPLIED,
};
```

### PR DESCRIPTION
**Pull request by axel-eck**
I created an event system for the server, so users can fetch for new events.

I also fixed a bug where the route to get channels would always return 404 Not Found

### CODE REVIEW

**Review by 6im0n — DISMISSED** [review]
Good for me ;)

File: tests/json/json_parser.c [hunk 1]
```diff
@@ -331,3 +331,54 @@ Test(json, safety_checks)
...
+
+    json_string_t *username = (json_string_t *) json_object_get(data, "username");
+    cr_assert(username != NULL);
+    cr_assert(username->base.type == JSON_OBJECT_TYPE_STRING);
+    cr_assert_str_eq(username->value, "marius");
+
+    json_destroy((json_t *) json);
+    free(json);
+    remove("../../tests/json/json_parse_array_of_objects_with_int.json");
+}
```
Comment (Marius-P1) [inline tests/json/json_parser.c]:
```suggestion
}

```

**Review by Marius-P1 — DISMISSED** [review]
:ok_hand:

**Review by 6im0n — DISMISSED** [review]

**Review by Marius-P1 — APPROVED** [review]
:ok_hand:

**Review by 6im0n — APPROVED** [review]
:+1:

**Pull request merged**